src/server/endpoints/profile-endpoint.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const tokensDb = require('../collections/tokens-db');
+const usersDb = require('../collections/users-db');
+const videosDb = require('../collections/videos-db');
+
+function getUserProfiles(req, res) {
+  if (req.get('Authorization') === undefined) {
+    res.status(400).json({'error': 'unAuthorization'});
+    return;
+  }
+  tokensDb.getToken(req.get('Authorization'), (tokenInfos) => {
+    if (tokenInfos._id === undefined) {
+      res.status(404).json({'error': 'not found'});
+      return;
+    }
+    usersDb.findUserInfoByUsername(req.params.username, (userInfos) => {
+      if (userInfos._id === undefined) {
+        res.status(404).json({'error': 'not found'});
+        return;
+      }
+      videosDb.getUploadVideosByUsername(req.params.username, (uploadVideos) => {
+        res.status(200).json({
+          'username': userInfos.username,
+          'email': userInfos.email,
+          'phoneNumber': userInfos['phone number'],
+          'fullName': userInfos['full name'],
+          'avatar': userInfos.avatar,
+          'subscribers': userInfos.subscribers,
+          'uploads': uploadVideos,
+        });
+      });
+    });
+  });
+}
+
+function modifyUserProfiles(req, res) {
+  if (req.get('Authorization') === undefined) {
+    res.status(400).json({'error': 'unAuthorization'});
+    return;
+  }
+  tokensDb.getToken(req.get('Authorization'), (tokenInfos) => {
+    if (tokenInfos._id === undefined) {
+      res.status(404).json({'error': 'not found'});
+      return;
+    }
+    usersDb.updateUserInfos(req.params.username, req.body, (updateResult) => {
+      if (updateResult === 'failed') {
+        res.status(400).json({'error': 'update failed'});
+        return;
+      }
+      res.set('location', '/api/profile/' + updateResult.username);
```

Comment:
we dont really use location in case of update. Only when we create a new resource. Here is no creation just modification